composablejs/packages/vesting-setup/src/main.ts

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ts-node
+/**
+ * This is the entry point for the vesting schedules initialization script.
+ *
+ * It creates vesting schedules based on `schedules.json` and `config.json`
+ */
+import "@composable/types/augment-api";
+import "@composable/types/augment-types";
+
+import { getDevWallets, sendAndWaitForSuccess, getNewConnection } from "@composable/utils";
+import { ApiPromise, Keyring } from "@polkadot/api";
+import { expect } from "chai";
+// This file contains the asset id and the `from` account.
+// TODO: Replace mock data
+import { config } from "./config.json";
+// This file contains the vesting schedules to be generated.
+// TODO: Replace mock data
+import { schedules } from "./schedules.json";
+
+/**
+ * Establishes connection.
+ *
+ * @returns {
+ * api: ApiPromise,
+ * keyring: Keyring,
+ * walletAlice: KeyringPair,
+ * }
+ */
+export async function connect(): Promise<{ api: ApiPromise; keyring: Keyring }> {
+  const endpoint = "ws://" + (process.env.ENDPOINT ?? "127.0.0.1:9988");
+  const { newClient, newKeyring } = await getNewConnection(endpoint);
+  return { api: newClient, keyring: newKeyring };
+}
+
+const main = async () => {
+  console.log("Composable Vesting Schedules Initialization");
+  console.debug("Connecting...");
+  // Establish connection to the node.
+  const { api, keyring } = await connect();
+
+  // Get wallets.
+  const { devWalletAlice } = getDevWallets(keyring);
+
+  // Create vested transfers.
+  const transactions = schedules.map(schedule => {
+    const scheduleInfo = api.createType("ComposableTraitsVestingVestingScheduleInfo", {
+      window: api.createType("ComposableTraitsVestingVestingWindow", {
+        BlockNumberBased: {
+          start: api.createType("u64", schedule.startBlock),
+          period: api.createType("u64", schedule.windowPeriod)
+        }
+      }),
+      periodCount: schedule.vestingPeriodCount,
+      perPeriod: api.createType("u128", schedule.perPeriodAmount)
+    });
+
+    return api.tx.vesting.vestedTransfer(config.address, schedule.beneficiary, config.asset, scheduleInfo);
+  });
+
+  // Execute transactions.
+  console.log("Creating schedules...");
+  const {
+    data: [result]
+  } = await sendAndWaitForSuccess(
```

Comment:
There's also an extra method called `sendWithBatchAndWaitForSuccess` for easy use.